lambda_functions/v2/functions/csv-forwarder/main.go

```diff
@@ -0,0 +1,169 @@
+package main
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type Lambda struct {
+	s3Client       s3iface.S3API
+	digidepsClient DigidepsClient
+}
+
+type DigidepsClient interface {
+	Post(url, contentType string, body io.Reader) (resp *http.Response, err error)
+}
+
+type S3EventRecord struct {
+	S3 struct {
+		Bucket struct {
+			Name string `json:"name"`
+		} `json:"bucket"`
+		Object struct {
+			Key string `json:"key"`
+		} `json:"object"`
+	} `json:"s3"`
+}
+
+type ObjectCreatedEvent struct {
+	S3EventRecords []S3EventRecord `json:"Records"`
+}
+
+func (l *Lambda) HandleEvent(event events.SQSEvent) (string, error) {
+	url := os.Getenv("DIGIDEPS_API_ENDPOINT")
+
+	if url == "" {
+		msg := "DIGIDEPS_API_ENDPOINT environment variable not set"
+		log.Print(msg)
+		return "", errors.New(msg)
+	}
+
+	input := s3.GetObjectInput{}
+	parsedEvent, err := SQSMessageParser(event)
+
+	if err != nil {
+		msg := fmt.Sprintf("Unable to parse SQS Message: %s", err.Error())
+		log.Print(msg)
+		return "", errors.New(msg)
+	}
+
+	input.Bucket = aws.String(parsedEvent.S3.Bucket.Name)
+	input.Key = aws.String(parsedEvent.S3.Object.Key)
+
+	resp, _ := l.s3Client.GetObject(&input)
+
+	defer resp.Body.Close()
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(resp.Body)
+
+	if err != nil {
+		msg := fmt.Sprintf("Unable to read file from S3: %s", err.Error())
+		log.Print(msg)
+		return "", errors.New(msg)
+	}
+
+	csvContents := buf.String()
+
+	encodedCSV := base64.StdEncoding.EncodeToString([]byte(csvContents))
+	postBody, err := json.Marshal(map[string]string{
+		"csv": encodedCSV,
+	})
+
+	if err != nil {
+		msg := fmt.Sprintf("Unable to marshal string to JSON: %s", err.Error())
+		log.Print(msg)
+		return "", errors.New(msg)
+	}
+
+	requestBody := bytes.NewBuffer(postBody)
+
+	response, err := l.digidepsClient.Post(url, "application/json", requestBody)
+
+	if err != nil {
+		msg := fmt.Sprintf("Unable to post to Digideps: %s", err.Error())
+		log.Print(msg)
+		return "", errors.New(msg)
+	}
+
+	if response.StatusCode != 202 {
+		return "", errors.New(fmt.Sprintf(
```

Comment:
We can just used `fmt.Errorf('...')` here instead of `errors.New(fmt.Sprintf('...'))`

Also errors strings should not be capitalised unless they begin with proper nouns or acronyms

So `Received unexpected status code...` should become `received unexpected status code...`